project/src/js/lichess/friends.js

```diff
@@ -1,8 +1,40 @@
-import { caseInsensitiveSort } from '../utils';
-
 var onlineFriends = [];
 
+var makeUser = function(name, isPlaying) {
+    return {
+      'name' : name,
+      'isPlaying' : isPlaying,
+    }
+};
+
+/** Compares usernames for equality, ignoring prefixed titles (such as GM) */
+var isSameUser = function(userId, name) {
+  var id = (name.indexOf(' ') >= 0) ? name.split(' ')[1] : name;
+  return id.toLowerCase() === userId;
+};
+
+var findByUsername = function(n) {
+  return onlineFriends.filter(function(u) {
+    return isSameUser(n.toLowerCase(), u.name);
+  })[0];
+}
+
+var setPlaying = function(userName, playing) {
+  var user = findByUsername(userName);
```

Comment:
Prefer `const` keyword here. I write the code in [ES2015](https://babeljs.io/docs/learn-es2015/).

Or... even better, you can rename the file to `friends.ts` and rewrite it in typescript! It's easy to learn, it's just a superset of javascript with types.
Currently ~20% of the code is in typescript, the more the better.